src/cocotb/share/lib/verilator/verilator.cpp

```diff
@@ -62,16 +81,19 @@ int main(int argc, char** argv) {
     VerilatedVpi::callCbs(cbStartOfSimulation);
 
 #if VM_TRACE
-    Verilated::traceEverOn(true);
 #if VM_TRACE_FST
     std::unique_ptr<VerilatedFstC> tfp(new VerilatedFstC);
-    top->trace(tfp.get(), 99);
-    tfp->open("dump.fst");
+    const char* traceFile = "dump.fst";
 #else
-    std::unique_ptr<VerilatedVcdC> tfp(new VerilatedVcdC);
-    top->trace(tfp.get(), 99);
-    tfp->open("dump.vcd");
+    std::unique_ptr<VerilatedvcdC> tfp(new VerilatedVcdC);
+    const char* traceFile = "dump.vcd";
```

Comment:
Any possibility that this can be made into an argument and the FST vs VCD also decided at runtime by looking at the extension?

I'm guessing we could also make `VERILATOR_SIM_DEBUG` a runtime flag, no?

Also I'm wondering if there is a reason not to always compile with `VM_TRACE`, `VM_COVERAGE` and use runtime switches to turn them on. Do they slow down the sim or the build considerably?

Reply:
Unfortunately the verilated C++ can only be no trace, vcd trace or fst trace.  There's no vcd or fst trace option.

An option to set `VERILATOR_SIM_DEBUG` while building does sound like a good idea.  Seems like a separate commit, but I can add it here is desired.

I have tried benchmarking trace vs no trace models over the years and have never been impressed enough with the performance gains to bother building no trace variants.  But other runner flavors seem to support this construct, so maybe  it's best to keep it?  As I mentioned, we don't plan to use `build()` in the Verilator runner so I don't have a strong preference.

Coverage does slow down the Verilator sims so I would not suggest adding that by default.

Reply:
> Unfortunately the verilated C++ can only be no trace, vcd trace or fst trace. There's no vcd or fst trace option.

So Verilator simply doesn't support compiling support for both in and only using the appropriate type? Odd.